### PR TITLE
aii-ks: fix swap noformat in ks not working under anaconda 7

### PR DESCRIPTION
**Pull request by kwaegema**

### CODE REVIEW

**Comment by Pansanel** [comment]
I will check the fix tomorrow (new CentOS7 install on the way)

File: aii-ks/src/main/perl/ks.pm [hunk 1]
```diff
@@ -536,7 +536,7 @@ EOF
 
         # EL7+ anaconda does not allow a preformatted / filesystem
```
Comment (stdweird) [inline aii-ks/src/main/perl/ks.pm]: adapt comment as well

**Comment by stdweird** [comment]
@kwaegema no unittest?

**Comment by kwaegema** [comment]
comment and unittest added. also tested to rollout node here

File: aii-ks/src/test/resources/kickstart_mounts.pan [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        "mount", true,
+        "mountpoint", "swap",
+        "preserve", true,
+        "format", false,
+        "mountopts", "auto",
+        "block_device", "partitions/sdb1",
+        "type", "swap",
+        "freq", 0,
+        "pass", 1
+        )
```
Comment (jrha) [inline aii-ks/src/test/resources/kickstart_mounts.pan]: This indentation is messy.

File: aii-ks/src/test/resources/kickstart_mounts.pan [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        "sdb1", nlist (
+            "holding_dev", "sdb",
+            "size", 100,
+            "type", "primary", # no defaults !
+            ),
+        )
+);
+
+"/system/filesystems" = list (
+    nlist (
```
Comment (jrha) [inline aii-ks/src/test/resources/kickstart_mounts.pan]: dict

File: aii-ks/src/test/resources/kickstart_mounts.pan [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+
+"/hardware/harddisks/sdb" = nlist(
+    "capacity", 4000,
+);
+
+"/system/blockdevices" = nlist (
+    "physical_devs", nlist (
+        "sdb", nlist ("label", "gpt")
+        ),
```
Comment (jrha) [inline aii-ks/src/test/resources/kickstart_mounts.pan]: minus four spaces

**Comment by jrha** [comment]
Mostly LGTM

**Comment by kwaegema** [comment]
@jrha indents fixed

File: aii-ks/src/test/perl/kickstart_mounts.t [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+my $fh = CAF::FileWriter->new("target/test/ks");
+# This module simply prints to the default filehandle.
+select($fh);
+
+my $ks = NCM::Component::ks->new('ks');
+my $cfg = get_config_for_profile('kickstart_mounts');
+
+NCM::Component::ks::ksmountpoints($cfg);
+like($fh, qr{^part swap --onpart sdb1 --fstype=swap --fsoptions='auto'}m, 'swap part ok for el7');
```
Comment (stdweird) [inline aii-ks/src/test/perl/kickstart_mounts.t]: needs to be stricter (e.g. no --noformat at end, so set `$`), and it's better to also use `unlike ($fh, qr{sdb1.*--noformat},...)`

**Comment by jrha** [comment]
No love for dict?

File: aii-ks/src/test/resources/kickstart_mounts.pan [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    "physical_devs", nlist (
+        "sdb", nlist ("label", "gpt")
+     ),
+    "partitions", nlist (
+        "sdb1", nlist (
+            "holding_dev", "sdb",
+            "size", 100,
+            "type", "primary", # no defaults !
+            ),
+        )
```
Comment (jrha) [inline aii-ks/src/test/resources/kickstart_mounts.pan]: This still has extra indents.

**Comment by jrha** [comment]
Lovely :heart_eyes:

**Comment by stdweird** [comment]
oneliner to fix systems with partioned swap
```bash
export `blkid -t TYPE=swap -o export`;grep $UUID /etc/fstab >& /dev/null; if [ $? -ne 0 ]; then echo fixing swap in fstab; cp /etc/fstab /etc/fstab.missingfstab; echo -e "UUID=$UUID\tswap\tswap\tdefaults\t0\t0" >> /etc/fstab; fi; /usr/lib/systemd/system-generators/systemd-fstab-generator; cp /tmp/*swap /run/systemd/generator; systemctl daemon-reload; systemctl start swap.target
```

**Pull request merged**